Makefile

```diff
@@ -1,13 +1,29 @@
-GOTOOLS = \
+GOTOOLS := \
```

Comment:
how to make this deterministic

Reply:
As discussed, the tools (external dependencies) added here should not change the way the binary gets built. (With some caveats.)
Making the brought-in tools secure (no malicious code introduced) is a different problem, not covered in this PR.
Here's a breakdown of tools and their impact on deterministic builds:
gox: gox can introduce flags to the build under the hood that might change the binary. We need to follow the changes in the gox repo to see if that happens. (Or deal with it after the fact.) Since the team has raised that we want to move to `xgo`, I wouldn't put much pressure on this. (We'll raise during the evaluation of xgo.)
glide: the tool that allows versioning for the internal libraries. Since it doesn't run the build process, the only impact I see is that it pulls down wrong versions - which is the malicious code category, we don't covere here.
ghr: ghr is run _after_ the build has finished. Also, it's part of the distribution process that should be separated from the source code and the build process. This will eventually be removed.
gometalinter: already versioned using gpkg.in.

Reply:
What @Greg-Szabo said, we shouldn't rely on gox or xgo or anything other than go by default.  I never want to run any of those on my computer.

Reply:
It's not a different problem.  Rather, it is the problem.

This PR introduces tools that we don't want to audit.

I will never run gox or xgo, and neither should most people.  We shouldn't trust them unless we audit their code too, and snapshot it.  I don't want to do that.

Lets use cross compiling maybe for automated testing, not for releasing, unless we audit these tools.

Reply:
How are we going to release without cross compiling?

Reply:
Go handles cross compiling. I just looked through gox, it's simple enough so we could use it if we thought it was better than for-looping with GOOS GOARCH set.  (See @melekes 's https://github.com/tendermint/tendermint/pull/1086 )

xgo on the other hand looks like a pain to audit.  We can avoid using xgo by avoiding the use of cgo.